src/plugins/intel_gpu/src/plugin/ops/moe.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "intel_gpu/op/moe_fused_compressed.hpp"
+#include "intel_gpu/plugin/common_utils.hpp"
+#include "intel_gpu/plugin/program_builder.hpp"
+#include "intel_gpu/primitives/moe_fused_compressed.hpp"
+
+namespace ov {
+namespace op {
+namespace internal {
+using MOEFusedCompressed = ov::intel_gpu::op::MOEFusedCompressed;
+}  // namespace internal
+}  // namespace op
+}  // namespace ov
+
+namespace ov::intel_gpu {
+
+static void CreateMOEFusedCompressedOp(ProgramBuilder& p, const std::shared_ptr<ov::intel_gpu::op::MOEFusedCompressed>& op) {
+    auto inputs = p.GetInputInfo(op);
+    const auto& config = op->get_config();
+    OPENVINO_ASSERT(inputs.size() == 11, "Inputs count of MOEFusedCompressed should be 11");
```

Comment:
ASSERT only works for DEBUG mode. 
You can use validate_inputs_count() for checking # inputs. 

BTW "should be 11" means that this only works for "asym weights with zp" right ?
(e.g., sym weight w/o zp should not be mapped) 
In that case, MOE itself should not be fused right?